fvm/src/gas/charge.rs

```diff
@@ -1,20 +1,24 @@
 // Copyright 2019-2022 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
+use std::borrow::Cow;
+
 use super::Gas;
 
 /// Single gas charge in the VM. Contains information about what gas was for, as well
 /// as the amount of gas needed for computation and storage respectively.
-pub struct GasCharge<'a> {
-    pub name: &'a str,
+#[derive(Clone, Debug)]
+pub struct GasCharge {
+    pub name: Cow<'static, str>,
```

Comment:
Why not a String? are we trying to enforce that price names must be known at compiletime?

Reply:
I'm trying to avoid allocating per gas-charge.

For some history:

- The string used to be `'static`.
- Then we made it take a lifetime, but that meant I couldn't store it.
- Now I'm making it a `Cow` so we can store it and _usually_ not have to allocate (unless the name comes from an actor itself).